stl/inc/ranges

```diff
@@ -3767,10 +3771,18 @@ namespace ranges {
             }
 
             _NODISCARD constexpr decltype(auto) operator*() const
-                noexcept(noexcept(_STD get<_Index>(*_Current))) /* strengthened */ {
+                noexcept(noexcept(_STD get<_Index>(*_Current))) /* strengthened */
+                requires is_reference_v<range_reference_t<_Base>> {
                 return _STD get<_Index>(*_Current);
             }
 
+            _NODISCARD constexpr decltype(auto) operator*() const noexcept(
+                is_nothrow_move_constructible_v<remove_cv_t<tuple_element_t<_Index, range_reference_t<_Base>>>> //
+                    && noexcept(_STD get<_Index>(*_Current))) /* strengthened */ {
+                using _ElemTy = remove_cv_t<tuple_element_t<_Index, range_reference_t<_Base>>>;
+                return static_cast<_ElemTy>(_STD get<_Index>(*_Current));

```

Comment:
With the return type replaced with `auto`, I don't think we need to spell the type out anymore:
```suggestion
                return _STD get<_Index>(*_Current);
```

Reply:
Ditto nope.